Xamarin.Forms.Platform.Android/Extensions/ImageViewExtensions.cs

```diff
@@ -29,17 +30,28 @@ internal static class ImageViewExtensions
 
 			ImageSource source = newImage?.Source;
 			Bitmap bitmap = null;
+			Drawable drawable = null;
+
 			IImageSourceHandler handler;
 
 			if (source != null && (handler = Internals.Registrar.Registered.GetHandlerForObject<IImageSourceHandler>(source)) != null)
 			{
-				try
+				if (handler is FileImageSourceHandler)
```

Comment:
Should check source and be done as
```
if (source is FileImageSource fileSource)
    imageView.Context.GetDrawable(fileSource);
```
This avoid extra cast checks and also if they did something fucky with their handlers we wont get a bad cast.

Reply:
@jassmith  So the thinking here was for backwards compatibility. FileImageSourceHandler is a sealed class so if the handler is of a different type that means the user registered their own handler to process the FileImageSource and my new code shouldn't be executed here..  Ideally we would just be changing IImageSourceHandler to return a Drawable and move all my code here into FileImageSourceHandler but since that would change the APIs I figured this fix would be a good performance gain for now without burdening the users

Reply:
Good point